internal/compiler/diagnostics.rs

```diff
@@ -91,14 +91,31 @@ impl SourceFileInner {
         Rc::new(Self { path, ..Default::default() })
     }
 
+    /// Returns a tuple with the line (starting at 1) and column number (starting at 0)
+    pub fn line_column(&self, offset: usize) -> (usize, usize) {
+        let line_offsets = self.line_offsets();
+        line_offsets.binary_search(&offset).map_or_else(
+            |line| {
+                if line == 0 {
+                    (1, offset)
+                } else {
+                    (line + 1, line_offsets.get(line - 1).map_or(0, |x| offset - x))
+                }
+            },
+            |line| (line + 1, 0),
+        )
+    }
+
     fn line_offsets(&self) -> &[usize] {
         self.line_offsets.get_or_init(|| {
             self.source
                 .as_ref()
                 .map(|s| {
                     s.bytes()
                         .enumerate()
-                        .filter_map(|(i, c)| if c == b'\n' { Some(i) } else { None })
+                        // Add the offset one past the '\n' into the index: That's the first char
+                        // of the new line!
+                        .filter_map(|(i, c)| if c == b'\n' { Some(i + 1) } else { None })
```

Comment:
the change looks correct at first glance, but that's strange that it worked before, because this should be tested. maybe the documenation of line_column is wrong?

Reply:
I believe this would add an offset of 1 to every line.
Remember we do `offset - x` in the `line_column` function

Reply:
So If i understand correctly, the comment in `line_column` was wrong because it used to be documented that the column starts as 0 while in fact, it started at 1 (except for the first line)?

I see you change the tests to adjust to that, but all the caller also need to adjust, this include the LSP, otherwise the diagnostic will be reported at the wrong place  (and there is no tests for that unfortunately)